cluster/addons/registry/README.md

```diff
@@ -0,0 +1,256 @@
+# Private Docker Registry in Kubernetes
+
+Kubernetes offers an optional private Docker registry addon, which you can turn
+on when you bring up a cluster or install later.  This gives you a place to
+store truly private Docker images for your cluster.
+
+## How it works
+
+The private registry runs as a `Pod` in your cluster.  It does not currently
+support SSL or authentication, which triggers Docker's "insecure registry"
+logic.  To work around this, we run a proxy on each node in the cluster,
+exposing a port onto the node (via a hostPort), which Docker accepts as
+"secure", since it is accessed by `localhost`.
+
+## Turning it on
+
+Some cluster installs (e.g. GCE) support this as a cluster-birth flag.  The
+`ENABLE_CLUSTER_REGISTRY` variable in `cluster/gce/config-default.sh` governs
+whether the registry is run or not.  To set this flag, you can specify
+`KUBE_ENABLE_CLUSTER_REGISTRY=true` when running `kube-up.sh`.  If your cluster
+does not include this flag, the following steps should work.  Note that some of
+this is cloud-provider specific, so you may have to customize it a bit.
+
+### Make some storage
+
+The primary job of the registry is to store data.  To do that we have to decide
+where to store it.  For cloud environments that have networked storage, we can
+use Kubernetes's `PersistentVolume` abstraction.  The following template is
+expanded by `salt` in the GCE cluster turnup, but can easily be adapted to
+other situations:
+
+<!-- BEGIN MUNGE: EXAMPLE registry-pv.yaml.in -->
+```
```

Comment:
Because they are yaml.in it's a little weird, but OK.  The example syncer
ignores it because it is not yaml, anyway.

On Thu, Oct 8, 2015 at 11:35 AM, Brendan Burns <notifications@github.com>
wrote:

> In cluster/addons/registry/README.md
> <https://github.com/kubernetes/kubernetes/pull/15182#discussion_r41550640>
> :
>
> > +`ENABLE_CLUSTER_REGISTRY` variable in `cluster/gce/config-default.sh` governs
> > +whether the registry is run or not.  To set this flag, you can specify
> > +`KUBE_ENABLE_CLUSTER_REGISTRY=true` when running `kube-up.sh`.  If your cluster
> > +does not include this flag, the following steps should work.  Note that some of
> > +this is cloud-provider specific, so you may have to customize it a bit.
> > +
> > +### Make some storage
> > +
> > +The primary job of the registry is to store data.  To do that we have to decide
> > +where to store it.  For cloud environments that have networked storage, we can
> > +use Kubernetes's `PersistentVolume` abstraction.  The following template is
> > +expanded by `salt` in the GCE cluster turnup, but can easily be adapted to
> > +other situations:
> > +
> > +<!-- BEGIN MUNGE: EXAMPLE registry-pv.yaml.in -->
> > +```
>
> tick-tick-tick yaml
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/15182/files#r41550640>.
>